File: nihal-files/OOPS_CONCEPTS/DigitalTv.java
package OOPS_CONCEPTS;

public class DigitalTv {
    
    private String brand;
    private String model;
    private int screenSize;
    private boolean isOn;
    private int channel;
    private int volume;

    //Constructor.....

    public DigitalTv(String brand, String model, int screenSize){
        this.brand = brand;
        this.model = model;
        this.screenSize = screenSize;

        this.isOn =false;
        this.channel = 1;
        this.volume = 10;
    }

    //Methods
 
    public void turnOn(){
        if(!isOn){
            isOn = true;
            System.out.println("welcome " + brand + "Tv is Turning On");
        }else{
            System.out.println(brand + "Tv is Already On");
        }
    }

    public  void turnOff(){
        if(isOn){
            isOn = false;  
            System.out.println(brand + "Tv is Turning Off...");

        }else{
            System.out.println(brand + "Tv is Already Off....");
        }
    }

    public void displayInfo(){
            System.out.println("brand : " + brand);
            System.out.println("Model : " + model);
            System.out.println("Screen Size : " + screenSize + "inch");

            if(isOn){
                System.out.println("Status : ON");
                System.out.println("Current Channel: " + channel);
                System.out.println("Current volume :" + volume);
            }else{
                System.out.println("Status : OFF");
            }
    }
// channel change
    public void changeChannel(int newchennal){
           if(isOn){
            channel = newchennal;
            System.out.println("The Chenl Changesd to " + channel);
           }else{
            System.out.println("Cannot change channel, the TV is off");
           }
    }
//Volume Adjustment
    public void adjustVolume(int amount){
        if(isOn){
            volume += amount;

             if (volume > 100) {
                volume = 100;
            } else if (volume < 0) {
                volume = 0;
            }

            System.out.println("Volume set to " + volume);
        } else {
            System.out.println("Cannot adjust volume, the TV is off.");
        }
    }

    
    
//Main Functions    
    public static void main(String[] args) {
        
        DigitalTv  tv = new DigitalTv("samsung", "SmartTv", 32);

        tv.displayInfo();
        // tv.turnOff();
        tv.turnOn();
        // tv.turnOff();
        tv.displayInfo();
        tv.changeChannel(54);
        tv.adjustVolume(0);
    }
}

File: damini-file/DigitalTv.py
class DigitalTv:

  def __init__(self,brand,model,screen_size,):
    self.brand=brand
    self.model=model
    self.screen_size=screen_size
    self.is_on = False
    self.channel=1
    self.volume= 10

  def turn_on(self):
    if not self.is_on:
      self.is_on=True
      print(f"Welcome! {self.brand} tv is turnning on.")
    else:
      print(f"{self.brand} TV is already on")

  def turn_off(self):
    if self.is_on:
      self.is_on = False
      print(f"{self.brand} TV is turninng off")
    else:
      print(f"{self.brand} TV is already off")

  def display_info(self):
    print(f"TV Brand: {self.brand},Model: {self.model},Screen size: {self.screen_size}")
    current_status= "on TV " if  self.is_on else "off TV" 
    print(f"Device: {self.brand}, Status:{current_status}") 


  def change_channel(self,new_channel):
    if self.is_on == True:
      # self.channel += 1
      self.channel = new_channel
      print(f"{self.brand} TV channel is changed to new channel {self.channel}")
    else:
      print(f"{self.brand}TV is off. turn it on")
  
  def adjust_volume(self, amount):
    if self.is_on:
        self.volume += amount

        if self.volume < 0:
            self.volume = 0
        elif self.volume > 100:
            self.volume = 100

        print(f"{self.brand} is ON, volume is now {self.volume}")
    else:
        print(f"{self.brand} TV is OFF, cannot adjust volume")
  
#create an object of digitalTv
tv_1=DigitalTv("LG","OLED C1" ,"48 INCH") 
# tv_2=DigitalTv("SONY","Bravia X90j","68 INCH")
# tv_3=DigitalTv("TCL","Roku tv","55 INCH")


# change channel and see the status of tv
tv_1.turn_on()
tv_1.change_channel(10)
tv_1.turn_off()
tv_1.change_channel(5)
tv_1.display_info()


##Now change the volume of tv and check the status of tv
tv_1.turn_on()
tv_1.adjust_volume(10)
tv_1.adjust_volume(-5)
tv_1.adjust_volume(-50)
tv_1.adjust_volume(150)


 




